### DISCUSSION TITLE
Restructure of legend entries

### DISCUSSION THREAD

**Discussion by MoBosse** [open-sdg/open-sdg#1401] [Ideas]
We have received feedback from some users that the legend entries are often confusing, especially when there are many entries with different subcategories. 
Is it possible to structure them a bit? Here are some suggestions: 

- Give each entry a new line
- Make them left-aligned
- Indentation per given subcategory
- Add a line to separate different categories

![grafik](https://user-images.githubusercontent.com/50657629/74936729-fc62af80-53ea-11ea-9765-f12d49863ad6.png)

I tried to realize some of it. See example [here](https://nachhaltige-entwicklung-deutschland.github.io/open-sdg-site-starter/en/11-1-b/). I like the result, but the coding is pretty poor (the indentation is based on adding spaces for each comma in the legend entry) so maybe someone has an idea how to achieve this in a more stable and elegant way.

**Comment by brockfanning** [comment]
@MoBosse 
I agree the legend is confusing, and your solution seems good to me. To give someone a headstart, maybe you could post a code snippet and/or link to the section of code that needs to be updated?

**Comment by MoBosse** [comment]
@brockfanning 
I think most of it would have to be done in the _legendCallback_ function [here]( https://github.com/open-sdg/open-sdg/blob/master/_includes/assets/js/indicatorView.js#L384).

To make the entries left-align _style="text-align: left; “_ can be added [here]( https://github.com/open-sdg/open-sdg/blob/master/_includes/assets/js/indicatorView.js#L385).

To show only one entry per line [this]( https://github.com/open-sdg/open-sdg/blob/master/_sass/layouts/_indicator.scss#L351) has to be changed to _display: block_

So far so easy. For the indentation some space must be added [here](https://github.com/open-sdg/open-sdg/blob/master/_includes/assets/js/indicatorView.js#L391) before the translated label. To do this properly, you need to check how many subcategories have a value for this dataset. One option would be to append a _list_of_subcategories_ to each dataset that looks e.g. like [sex, age, country, ...]. The length of this list gives the width of the space to add before the label. Unfortunately I don`t know where the _dataset_ that is given to _legendCallback_ is defined and how to create this _list_of_subcategories_.

Regarding the separation between different categories: I think, here another problem has to be solved first, namely the correct sorting of the entries by subcategory. If I'm not mistaken, the sorting at the moment is such that the _headline data_ comes first and the rest follows in alphabetical order.
I found this on the UK page as a good example of how that looks like:

![grafik](https://user-images.githubusercontent.com/50657629/77303238-e6e6eb00-6cf2-11ea-99ad-a882ca96bfd1.png)

I would have expected the order to be structured like the dropdown-menues. In this case:

- England and Wales
  - England
    - England, East
............................................
  - 20 - 24
  - Under 20
............................................
  - 0 - 3
  - Not started

If you have the above mentioned _list_of_subcategories_, I don't think it's that difficult anymore to insert the lines in the right place. But this only works if the order of the entries is correct.

**Comment by brockfanning** [comment]
User research planned in 2021.

**Comment by LucyGwilliamAdmin** [comment]
This should be looked alongside showing union of disaggregations on charts
